### DISCUSSION TITLE
Implementation of getNativeViews on iOS

### DISCUSSION THREAD

**Discussion by pszklarska** [leancodepl/patrol#1403] [Ideas]
Hi! 

I noticed that [NativeAutomator.getNativeViews](https://pub.dev/documentation/patrol/latest/patrol/NativeAutomator/getNativeViews.html) is currently not implemented on iOS and throws `getNativeViews() is not supported on iOS`. 

My project could really make use of that method in tests and I'd be happy to file a PR, so I wanted to gather some feedback first. Have you encountered any issues with getting a list of native views on iOS? I noticed a similar thing is already done in `Automator.swift`, for example to tap on the view, so (in theory) it should be possible:
https://github.com/leancodepl/patrol/blob/ccd821794fd5f7ea2c606fd3307d491cd7d1a86d/packages/patrol/ios/Classes/AutomatorServer/Automator.swift#L60-L61
Is there anything else that should be considered to add this functionality? 

Btw, I've been using patrol for a while now and I'm very impressed by this framework, so kudos to the whole team!

**Comment by bartekpacia** [comment]
Hi 👋🏻 

I agree that `NativeAutomator.getNativeViews()` is useful and it's a pity that it's not implemented on iOS.

> Have you encountered any issues with getting a list of native views on iOS?

Honestly: at this point, I don't remember anymore :) It was implemented almost a year ago on Android as a quick workaround for #499. But it's definitely useful.

Using `app.descendants(matching: .any)` sounds like a good starting point.

> My project could really make use of that method in tests and I'd be happy to file a PR.

That's great! [Here](https://github.com/leancodepl/patrol/blob/ec321c39b5b7c078937d97ffcdd40bc422c85c7b/packages/patrol/android/src/main/kotlin/pl/leancode/patrol/Automator.kt#L142-L147)'s how it's done on Android, fwiw.

> Is there anything else that should be considered to add this functionality?

It'd be cool to have a test for it, but we can take care of it after you get it to work.

> Btw, I've been using patrol for a while now and I'm very impressed by this framework, so kudos to the whole team!

Thank you so very much for kind words :) It's really motivating!

**Comment by bartekpacia** [comment]
The servers and clients we use for communication are generated from [this protobuf file](https://github.com/leancodepl/patrol/blob/master/contracts.proto) using [this script](https://github.com/leancodepl/patrol/blob/master/gen_protobufs).

**Comment by bartekpacia** [comment]
Done in #1553 and released in [`v2.2.0`](https://pub.dev/packages/patrol/changelog#220).
